packages/synapse-react-client/src/components/SynapseChat/SynapseChat.tsx

```diff
@@ -245,13 +261,48 @@ export const SynapseChat: React.FunctionComponent<SynapseChatProps> = ({
               )
             })}
             {pendingInteraction && (
-              <SynapseChatInteraction
-                userMessage={pendingInteraction.userMessage}
-                chatResponseText={pendingInteraction.chatResponseText}
-                chatErrorReason={pendingInteraction.chatErrorReason}
-                progressMessage={currentProgressMessage}
-                scrollIntoView
-              />
+              <>
+                <SynapseChatInteraction
+                  userMessage={pendingInteraction.userMessage}
+                  chatResponseText={pendingInteraction.chatResponseText}
+                  chatErrorReason={pendingInteraction.chatErrorReason}
+                  scrollIntoView
+                />
+                <Box
+                  sx={{
+                    display: 'flex',
+                    flexDirection: 'column',
+                    justifyContent: 'center',
+                  }}
+                >
+                  <SynapseSpinner size={40} />
+                  {/* Show the current message, as well as the full trace log in a tooltip */}
+                  {traceEvents && traceEvents.page && (
```

Comment:
Currently we ask for the full page of trace events (every second), but we could technically set the timestamp of the message we saw last and store them in another state variable.  Current implementation surfaces the latest trace event message, and surfaces the full (most recent) page of trace events in the tooltip